src/containers/shared/components/Transaction/OracleSet/parser.ts

```diff
@@ -0,0 +1,26 @@
+import { UInt64 } from 'ripple-binary-codec/dist/types'
+import { convertHexToString } from '../../../../../rippled/lib/utils'
+import { OracleSet } from './types'
+
+export function parser(tx: OracleSet) {
+  const priceDataSeries = tx.PriceDataSeries.map((priceDataObj) => ({
+    tradingPair: `${priceDataObj.PriceData.BaseAsset}/${priceDataObj.PriceData.QuoteAsset}`,
+
+    assetPrice:
+      priceDataObj.PriceData.AssetPrice && priceDataObj.PriceData.Scale
+        ? (
+            Number(UInt64.from(priceDataObj.PriceData.AssetPrice).valueOf()) /
+            10 ** priceDataObj.PriceData.Scale
+          ).toFixed(5)
+        : undefined,
+  }))
+
+  return {
+    oracleDocumentID: tx.OracleDocumentID,
+    provider: convertHexToString(tx.Provider),
+    uri: convertHexToString(tx.URI),
+    lastUpdateTime: tx.LastUpdateTime,
```

Comment:
Why not do the conversion here instead?

Reply:
It was because the `useLanguage` hook used within `localizeDate()` has to be used inside a React component or hook, but `parser` is a regular function so eslint would complain about it:

```
React Hook "useLanguage" is called in function "parser" that is neither a React function component nor a custom React Hook function. React component names must start with an uppercase letter. React Hook names must start with the word "use".
```